morgoth/addons/api/serializers.py

```diff
@@ -21,21 +31,46 @@ class Meta:
 
 
 class AddonGroupSerializer(serializers.ModelSerializer):
-    addons = serializers.PrimaryKeyRelatedField(many=True, queryset=Addon.objects.all(),
-                                                required=False)
+    addons = AddonSerializer(many=True, read_only=True)
+    addon_ids = serializers.ListField(required=False, write_only=True)
+    built_in_addons = AddonSerializer(many=True, read_only=True)
+    qa_addons = AddonSerializer(many=True, read_only=True)
+    shipped_addons = AddonSerializer(many=True, read_only=True)
+    browser_version = VersionField()
 
     class Meta:
         model = AddonGroup
         fields = (
             'id',
             'browser_version',
             'addons',
+            'addon_ids',
             'built_in_addons',
             'qa_addons',
             'shipped_addons',
         )
-        read_only_fields = (
-            'built_in_addons',
-            'qa_addons',
-            'shipped_addons',
-        )
+
+    def get_addons(self, obj):
```

Comment:
This API is deprecated, I think, and I think just assigning the serializer above like you did accomplishes the same thing.